tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_selectors.ts

```diff
@@ -111,3 +114,72 @@ export const getVisibleExecutionDigests = createSelector(
     return digests;
   }
 );
+
+export const getFocusedExecutionIndex = createSelector(
+  selectDebuggerState,
+  (state: DebuggerState): number | null => {
+    return state.executions.focusIndex;
+  }
+);
+
+/**
+ * Get the display index of the execution digest being focused on (if any).
+ */
+export const getFocusedExecutionDisplayIndex = createSelector(
+  selectDebuggerState,
+  (state: DebuggerState): number | null => {
+    if (state.executions.focusIndex === null) {
+      return null;
+    }
+    const {focusIndex, scrollBeginIndex, displayCount} = state.executions;
+    if (
+      focusIndex < scrollBeginIndex ||
+      focusIndex >= scrollBeginIndex + displayCount
+    ) {
+      return null;
+    }
+    return focusIndex - scrollBeginIndex;
+  }
+);
+
+export const getLoadedExecutionData = createSelector(
+  selectDebuggerState,
+  (state: DebuggerState): {[index: number]: Execution} =>
+    state.executions.executionData
+);
+
+export const getLoadedStackFrames = createSelector(
+  selectDebuggerState,
+  (state: DebuggerState): StackFramesById => state.stackFrames
+);
+
+export const getFocusedExecutionData = createSelector(
+  selectDebuggerState,
+  (state: DebuggerState): Execution | null => {
+    const {focusIndex, executionData} = state.executions;
+    if (focusIndex === null || executionData[focusIndex] === undefined) {
+      return null;
+    }
+    return executionData[focusIndex];
+  }
+);
+
+export const getFocusedExecutionStackFrames = createSelector(
```

Comment:
I find the behavior at L180 quite interesting. Can you leave a comment on the contract (i.e., if any stackframe data is missing, this will return `null` for the whole thing)? Also, what is an expected behavior in that case?